### DISCUSSION TITLE
Global Theme Provider

### DISCUSSION THREAD

**Discussion by cfluke-twilio** [twilio-labs/paste#950] [Ideas] [answered]
Hey all, is there any plan to add a global theme overrides. 
I see that a themeprovider does exist, but it looks like each child element will need to implement that theme style choice.
With a global theme override, I can say theme all buttons that are a certain variant and/or size with these css properties.

In other ui frameworks, there is a global override option.
Examples: 
[Material-UI](https://material-ui.com/customization/globals/)
[Chakra-UI](https://chakra-ui.com/docs/theming/customize-theme#customizing-component-styles)
[Semantic-UI](http://learnsemantic.com/themes/overview.html#css-overrides)

**Comment by SiTaggart** [comment]
Hi @cfluke-twilio 

There is a proposal for adding a customization layer to Paste currently under consideration. If all goes to plan it will get added to Paste next year sometime. That is not a guarantee though, based on budget, resources, planning etc.

The architectural blueprint can be found https://docs.google.com/document/d/10ovjW3EArixaMQMg-tChvUU7MR7r4nn5ary2oyjDpME/

**Comment by SiTaggart** [comment, marked as answer]
Hi @cfluke-twilio 

As part of Tweek week, the team created an quick easy way to theme Paste. Thought this might be of interest to you.

https://paste-theme-designer.vercel.app/edit

The tool can then export a theme as a JSON object. You can then apply that theme to a new Provider that is now part of the system. A demo can be found here: https://codesandbox.io/s/paste-customization-provider-qiiej?file=/src/index.tsx

We haven't documented any of this yet as it's a tweek project, but we'd be curious to start hearing feedback.
